datasafe-rest-impl/src/main/java/de/adorsys/datasafe/rest/impl/controller/VersionController.java

```diff
@@ -0,0 +1,131 @@
+package de.adorsys.datasafe.rest.impl.controller;
+
+import de.adorsys.datasafe.business.impl.service.VersionedDatasafeServices;
+import de.adorsys.datasafe.encrypiton.api.types.UserID;
+import de.adorsys.datasafe.encrypiton.api.types.UserIDAuth;
+import de.adorsys.datasafe.encrypiton.api.types.keystore.ReadKeyPassword;
+import de.adorsys.datasafe.metainfo.version.impl.version.types.DFSVersion;
+import de.adorsys.datasafe.types.api.actions.ListRequest;
+import de.adorsys.datasafe.types.api.actions.ReadRequest;
+import de.adorsys.datasafe.types.api.actions.RemoveRequest;
+import de.adorsys.datasafe.types.api.actions.WriteRequest;
+import de.adorsys.datasafe.types.api.resource.*;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.util.StreamUtils;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+import static org.springframework.http.MediaType.APPLICATION_OCTET_STREAM_VALUE;
+
+@Slf4j
+@RestController
+@RequestMapping
+@RequiredArgsConstructor
+public class VersionController {
+
+    private final VersionedDatasafeServices versionedDatasafeServices;
+
+    /**
+     * lists latest versions of files in user's private space.
+     */
+    @GetMapping(value = "/versioned/{path:.*}", produces = APPLICATION_JSON_VALUE)
+    public List<String> listVersionedDocuments(@RequestHeader String user,
+                                      @RequestHeader String password,
+                                      @PathVariable(required = false) String path) {
+        UserIDAuth userIDAuth = new UserIDAuth(new UserID(user), new ReadKeyPassword(password));
+        path = "./" + Objects.toString(path, "");
+        List<String> documentList = versionedDatasafeServices.latestPrivate().listWithDetails(ListRequest.forDefaultPrivate(userIDAuth, path))
+                .map(e -> e.absolute().getResource().decryptedPath().getPath())
+                .collect(Collectors.toList());
+        log.debug("List for path {} returned {} items", path, documentList.size());
+        return documentList;
+    }
+
+    /**
+     * reads latest version of file from user's private space.
+     */
+    @SneakyThrows
+    @GetMapping(value = "/versioned/{path:.*}", produces = APPLICATION_OCTET_STREAM_VALUE)
+    public void readVersionedDocument(@RequestHeader String user,
+                             @RequestHeader String password,
+                             @PathVariable String path,
+                             HttpServletResponse response) {
+        UserIDAuth userIDAuth = new UserIDAuth(new UserID(user), new ReadKeyPassword(password));
+        PrivateResource resource = BasePrivateResource.forPrivate("./" + path);
```

Comment:
You can use path directly instead of `"./" + path`

Reply:
removed